tests/tests/test_mtls.py

```diff
@@ -123,33 +122,13 @@ def hsm_setup(self, pin, ssl_engine_id, device):
 
     def hsm_get_key_uri(self, pin, ssl_engine_id, device):
         pt11tool_output = device.run(
-            "export SOFTHSM2_CONF=/softhsm/softhsm2.conf; p11tool --login --provider=/usr/lib/softhsm/libsofthsm2.so --set-pin="
-            + pin
-            + " --list-all-privkeys"
+            f"p11tool --login --provider=/usr/lib/softhsm/libsofthsm2.so --set-pin={pin} --list-all-privkeys"
         ).rstrip("\n")
         key_uri = re.search(r"URL:\s(.*)", pt11tool_output).group(1)
         key_uri = key_uri + ";pin-value=" + pin
 
-        device.run("cp /etc/ssl/openssl.cnf /etc/ssl/openssl.cnf.backup")
-        device.run(
-            'echo -ne "[openssl_init]\nengines=engine_section\n\n[engine_section]\npkcs11 = pkcs11_section\n\n[pkcs11_section]\nengine_id = '
-            + ssl_engine_id
-            + '\nMODULE_PATH = /usr/lib/softhsm/libsofthsm2.so\ninit = 0\n" >> /etc/ssl/openssl.cnf'
-        )
-        device.run(
-            'sed -i.backup -e "/\\[Service\\]/ a Environment=SOFTHSM2_CONF=/softhsm/softhsm2.conf" /lib/systemd/system/%s.service'
-            % device.get_client_service_name()
-        )
```

Comment:
ok, I do not understand this: if you dont configure the engine, then from what do you load the key?

Reply:
that cannot be. the engine id is arbitrary and there are many ways to give it in the config, how can it be in the code?

Reply:
Yes, but there is a default directory, in which it loads modules from

Reply:
You can override this with the config if you so desire of course

Reply:
Also remember, that we are not testing `Engines` now, but `Providers`

Reply:
I used this as my guide:

https://github.com/openssl/openssl/blob/master/README-PROVIDERS.md#loading-providers

And went for the explicit approach. Also please remember that this is not an `Engine` test. They don't exist for us in OpenSSLv3 which is what we are testing here.

Reply:
I asked a question on Mender-hub also, to see what our users think about this :)

https://hub.mender.io/t/pkcs11-hsm-support-for-authentication-authorization-and-signature-verification/886/18

Reply:
I was not aware that we are deprecating ENgines in the client in this way.

Reply:
Well, it's OpenSSL deprecating it. Not us :smile: 

We're just not enabling deprecated API's atm.